### PR TITLE
impl `num_traits::Num` and `num_traits::Signed` for `Decimal`

### PR DESCRIPTION
**Pull request by MathisWellmann**

### CODE REVIEW

File: src/decimal.rs [hunk 1]
```diff
@@ -101,6 +101,42 @@ where
...
+impl<I, const D: u8> num_traits::Num for Decimal<I, D> 
+where
+    I: SignedScaledInteger<D>,
+{
+    type FromStrRadixErr = &'static str;
+
+    fn from_str_radix(str: &str, radix: u32) -> Result<Self, Self::FromStrRadixErr> {
+        Ok(Self(I::from_str_radix(str, radix).map_err(|_| "Cannot parse from str")?))
+    }
+}
```
Comment (OliverNChalk) [inline src/decimal.rs]: I think this conflicts with the current to/from string impl that parses decimal format i.e. `1.250`

**Review by OliverNChalk — CHANGES_REQUESTED** [review]
Signed change looks good, not sure about the `from_str_radix` though, what's the use-case need for this + thoughts on how it differs from decimal representation?

**Comment by MathisWellmann** [comment]
For the `Signed` implementation it is required to implement `Num` as well, which is already a trait bound on all `I: ScaledInteger<D>`, because that requires `PrimInt` which requires `Num`. So my thinking was to just pass the functionality already in `I` through  in the `from_str_radix` implementation. On second thought, we must ensure that `radix == 10`, because thats the only numeric basis that makes sense for a decimal crate. I'll push some appropriate changes tomorrow.

File: src/decimal.rs (hunk 1 above)
Comment (MathisWellmann) [reply src/decimal.rs:113]: The implementations of `<I as FromStr>::parse_str()` and `<I as Num>::from_str_radix` for base 10 numbers (`radix == 10`) will be the same output, as the implementation of `Num::from_str_radix` actually just calls the standard methods of the primitive types, e.g: (https://doc.rust-lang.org/std/primitive.i32.html#method.from_str_radix).
Here is where `num_traits` implements `Num::from_str_radix` for the primitives: https://github.com/rust-num/num-traits/blob/022f250f7c12955166dd2ebe776f74e1707312a2/src/lib.rs#L169
So when implementing this method for `Decimal`, we can just use the implementation of the generic type `I`.

Comment (MathisWellmann) [reply src/decimal.rs]: I think you're right. It should use `Decimal::from_str` inside the `from_str_radix` method, but it also must ensure that `radix == 10` as different numeric bases wouldn't make much sense for a decimal crate.

**Comment by MathisWellmann** [comment]
I've change the implementation of `from_str_radix`, making sure that `radix == 10` and then using the `from_str` implementation already present.

File: src/decimal.rs [hunk 2]
```diff
@@ -101,6 +101,45 @@ where
...
+where
+    I: SignedScaledInteger<D>,
+{
+    type FromStrRadixErr = ParseDecimalError<I>;
+
+    fn from_str_radix(str: &str, radix: u32) -> Result<Self, Self::FromStrRadixErr> {
+        if radix != 10 {
+            return Err(ParseDecimalError::RadixMustBe10)
+        }
+        <Self as std::str::FromStr>::from_str(str)
```
Comment (OliverNChalk) [inline src/decimal.rs]:
```suggestion
        }

        <Self as std::str::FromStr>::from_str(str)
```

**Review by OliverNChalk — APPROVED** [review]

**Pull request merged**